spec.md

```diff
@@ -240,6 +240,25 @@ Messages can be delivered through various industry standard protocol (e.g. HTTP,
 AMQP, MQTT, SMTP), open-source protocols (e.g. Kafka, NATS), or
 platform/vendor specific protocols (AWS Kinesis, Azure Event Grid).
 
+## Type System
+
+The following abstract data types are available for use in attributes.
+
+- `String` - Sequence of printable Unicode characters.
+- `Binary` - Sequence of bytes.
+- `Map` - `String`-indexed dictionary of `Object`-typed values
+- `Object` - Either a `String`, or a `Binary`, or a `Map`
```

Comment:
The term `Object` is a bit confusing, since it has a very specific meaning in JavaScript and JSON will be a common encoding

Reply:
The [closest JSON equivalent for this is "value"](https://tools.ietf.org/html/rfc7159#section-3), which I would find even more confusing for mapping into other type systems. 

Since JSON and JavaScript are dynamically typed; there's simply no explicit "could be anything" type. TypeScript therefore [introduces "Any"](http://www.typescriptlang.org/docs/handbook/basic-types.html) 

`Object` is the generic "could be anything" base type in Java, C#, and Python, Golang has an Object Interface, etcetc. 

I'd prefer to align with the concepts of a lot of common languages rather than go out of the way of a particular language, especially also because this is [explicitly addressed in the JSON binding](https://github.com/cloudevents/spec/pull/148/files#diff-541c0c5dfe73eb8ec79427b6bab66748R61)

Reply:
My big issue was that we seemed to take a step back when Maps no longer specified what their value-type was. Is there a reason not to templatize Map still?

Reply:
I prefer `any` or `variant` to `Object`